src/pivot-table/hooks/use-grid-height.ts

```diff
@@ -0,0 +1,36 @@
+import type { HeadersData, LeftDimensionData, PageInfo, Rect, TopDimensionData } from "../../types/types";
+import { GRID_BORDER } from "../constants";
+import { useStyleContext } from "../contexts/StyleProvider";
+
+interface GridHeightHook {
+  pageInfo: PageInfo;
+  headersData: HeadersData;
+  topDimensionData: TopDimensionData;
+  leftDimensionData: LeftDimensionData;
+  tableRect: Rect;
+}
+
+export default function useGridHeight({
+  pageInfo,
+  headersData,
+  topDimensionData,
+  leftDimensionData,
+  tableRect,
+}: GridHeightHook) {
+  const { headerCellHeight, contentCellHeight } = useStyleContext();
+
+  const totalDataHeight = pageInfo.rowsOnCurrentPage * contentCellHeight;
+  const headerGridHeight = headerCellHeight * headersData.size.y;
+  const containerHeight = totalDataHeight + headerGridHeight;
+
+  // Top grid should always have height to support cases when there is no top data but it need to occupy space to currecly render headers
+  const topGridHeight = headerCellHeight * Math.max(topDimensionData.rowCount, 1);
+  const leftGridHeight = Math.min(tableRect.height - headerGridHeight - GRID_BORDER, totalDataHeight);
+  const dataGridHeight = Math.min(tableRect.height - topGridHeight - GRID_BORDER, totalDataHeight);
+
+  const rowsCanFitInTableViewPort = Math.floor(tableRect.height / contentCellHeight);
+  const rowsInCurrentPage = Object.values(Object.values(leftDimensionData.grid).at(-1) || {}).length;
+  const showLastBottomBorder = rowsInCurrentPage < rowsCanFitInTableViewPort;
```

Comment:
```suggestion
  const showLastBottomBorder = pageInfo.rowsOnCurrentPage < rowsCanFitInTableViewPort;
```

Reply:
so we shouldn't use the row count thing on the line above?

Reply:
No pageInfo already have that value

Reply:
WAIT!

Reply:
Let me double check this.

Reply:
`const rowsInCurrentPage = Object.values(Object.values(leftDimensionData.grid).at(-1) || {}).length;` is actually "number of that have been fetched". Which is working here but not really want we want.

Please go ahead and use `pageInfo.rowsOnCurrentPage`.

Reply:
nice, that simplifies things